test/mpi/test_mpi.cc

```diff
@@ -304,6 +305,16 @@ int ucc_coll_inplace_supported(ucc_coll_type_t c)
     }
 }
 
+int ucc_coll_persistent_supported(ucc_coll_type_t c)
+{
+    switch(c) {
+    case UCC_COLL_TYPE_BARRIER:
```

Comment:
@shimmybalsam i had another look. Lets add support for barrier as well. It looks easy. In the test_barrier.cc, if "persistent" we just don't need to re-init barrier request:
```
 82       if (i < size - 1) {
  83           UCC_CHECK(ucc_collective_finalize(req));
  84            UCC_CHECK(ucc_collective_init(&args, &req, team.team));
  85      }
```
The lines above should be skipped in persistent case. this looks all .

Reply:
added